com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs

```diff
@@ -0,0 +1,684 @@
+            public TextureHandle previousHistoryBuffer0;
+            public TextureHandle currentHistoryBuffer0;
+            public TextureHandle previousHistoryBuffer1;
+            public TextureHandle currentHistoryBuffer1;
+            public TextureHandle intermediateBuffer0;
+            public TextureHandle intermediateBuffer1;
+            public TextureHandle intermediateBufferDepth0;
+            public TextureHandle intermediateBufferDepth1;
+        }
+
+        private bool EvaluateVolumetricCloudsHistoryValidity(HDCamera hdCamera)
+        {
+            // Evaluate the history validity
+            return hdCamera.EffectHistoryValidity(HDCamera.HistoryEffectSlot.VolumetricClouds, false, false);
+        }
+
+        private void PropagateVolumetricCloudsHistoryValidity(HDCamera hdCamera)
+        {
+            hdCamera.PropagateEffectHistoryValidity(HDCamera.HistoryEffectSlot.VolumetricClouds, false, false);
+        }
+
+        TextureHandle TraceVolumetricClouds(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle colorBuffer, TextureHandle depthPyramid, TextureHandle motionVectors, TextureHandle volumetricLighting, HDUtils.PackedMipChainInfo info)
+        {
+            using (var builder = renderGraph.AddRenderPass<VolumetricCloudsData>("Generating the rays for RTR", out var passData, ProfilingSampler.Get(HDProfileId.VolumetricClouds)))
+            {
+                builder.EnableAsyncCompute(false);
+                VolumetricClouds settings = hdCamera.volumeStack.GetComponent<VolumetricClouds>();
+
+                passData.parameters = PrepareVolumetricCloudsParameters(hdCamera, settings, info, false, EvaluateVolumetricCloudsHistoryValidity(hdCamera));
+                passData.colorBuffer = builder.ReadTexture(builder.WriteTexture(colorBuffer));
+                passData.depthPyramid = builder.ReadTexture(depthPyramid);
+                passData.motionVectors = builder.ReadTexture(motionVectors);
+                passData.volumetricLighting = builder.ReadTexture(volumetricLighting);
+                passData.scatteringFallbackTexture = renderGraph.defaultResources.blackTexture3DXR;
+
+                passData.currentHistoryBuffer0 = renderGraph.ImportTexture(RequestCurrentVolumetricCloudsHistoryTexture0(hdCamera));
+                passData.previousHistoryBuffer0 = renderGraph.ImportTexture(RequestPreviousVolumetricCloudsHistoryTexture0(hdCamera));
+                passData.currentHistoryBuffer1 = renderGraph.ImportTexture(RequestCurrentVolumetricCloudsHistoryTexture1(hdCamera));
+                passData.previousHistoryBuffer1 = renderGraph.ImportTexture(RequestPreviousVolumetricCloudsHistoryTexture1(hdCamera));
+
+                passData.intermediateBuffer0 = builder.CreateTransientTexture(new TextureDesc(Vector2.one * 0.5f, true, true)
+                    { colorFormat = GraphicsFormat.R16G16B16A16_SFloat, enableRandomWrite = true, name = "Temporary Clouds Lighting Buffer 0" });
```

Comment:
all the lighting for cloud, isn't this suppose to reuse the current color buffer format?
GraphicsFormat.R16G16B16A16_SFloat
Guess we could follow it, or maybe give the option to use 111110f in the settings? it is a lot of fat texture.

Reply:
Not it cannot, the alpha channel is used